pkg/cloud/externalconn/utils/connection_utils.go

```diff
@@ -32,11 +28,8 @@ const markerFile = "crdb_external_storage_location"
 // CheckExternalStorageConnection writes a sentinel file, lists the file, and reads the file
 // back. This serves as a sanity check that the external connection represents
 // an ExternalStorage resource that can be connected and interacted with.
-func CheckExternalStorageConnection(
-	ctx context.Context, execCfg interface{}, user username.SQLUsername, uri string,
-) error {
-	cfg := execCfg.(*sql.ExecutorConfig)
-	es, err := cfg.DistSQLSrv.ExternalStorageFromURI(ctx, uri, user)
+func CheckExternalStorageConnection(ctx context.Context, env cloud.KMSEnv, uri string) error {
```

Comment:
Why can't we pass in a `ExternalConnEnv` here instead of a cloud.KMSEnv? It seems a little strange to me to be passing around a KMS environment in a method that validates non-KMS connection URIs. From what I can tell passing in a `ExternalConnEnv` should give us a handle to all the things we need?